buildingmotif/label_parsing/combinators.py

```diff
@@ -33,6 +33,17 @@ def __call__(self, target: str) -> List[TokenResult]:
             )
         ]
 
+    def to_dict(self):
```

Comment:
I wonder if the `Parser` super class could automatically generate these. 

The super constructor could take the `**kwargs` and just dump those into a dict to be jsonified and have the `name` populated through a similar method. you already instantiate the classes by expanding the `**args` so it should mostly work.